abnf/odata-abnf-construction-rules.txt

```diff
@@ -564,12 +564,15 @@ collectionPathExpr = count [ OPEN expandCountOption *( SEMI expandCountOption )
                    / "/" allExpr
                    / "/" boundFunctionExpr
                    / "/" annotationExpr
+                   / "/" indexExpr

```

Comment:
If prefer the less invasive variant
```
$orderby=components($orderby=name;$index=0)/name
```
so we need not extend the `collectionPathExpr` rule.

Reply:
This will also require extending the path syntax, allowing nested query options - at least `$orderby` and `$index`.